conda/core/solve.py

```diff
@@ -25,18 +27,34 @@
 from ..common.constants import NULL
 from ..common.io import Spinner, dashlist, time_recorder
 from ..common.path import get_major_minor_version, paths_equal
-from ..exceptions import PackagesNotFoundError, SpecsConfigurationConflictError, UnsatisfiableError
+from ..exceptions import (PackagesNotFoundError, SpecsConfigurationConflictError,
+                          UnsatisfiableError, CondaSolverTimeoutError)
 from ..history import History
 from ..models.channel import Channel
 from ..models.enums import NoarchType
 from ..models.match_spec import MatchSpec
 from ..models.prefix_graph import PrefixGraph
 from ..models.version import VersionOrder
 from ..resolve import Resolve
+from ..common.signals import raise_timeout
+
 
 log = getLogger(__name__)
 
 
+@contextmanager
+def solver_timeout(time):
```

Comment:
unfortunately this approach does not work for windows or python2

Reply:
on the bright side, we don't need to worry about supporting python 2 anymore!

Reply:
could maybe use something like https://dreamix.eu/blog/webothers/timeout-function-in-python-3 instead